### PR TITLE
Upg: stop fetching feature flags on all requests, do it on-demand

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Querying the feature flag is the sixth query that cost us the most CPU according to [GCP insight](https://console.cloud.google.com/sql/instances/dust-api-db/insights;database=dust-front;duration=P1D;sort_by=CPU/executed?project=or1g1n-186209).

It's very fast but done 9 millions times / day.

This refactor the code to fetch the feature flags only when needed.

If it's efficient, we should then look into the other query done 9 millions times (active subscription + plan), and then the other ones done 5-3 millions times (such as fetching the groups, the memberships etc..).

## Risk

I tested locally. Worst case, the feature behind feature flags are disabled.

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/lib/auth.ts [hunk 1]
```diff
@@ -233,23 +223,13 @@ export class Authenticator {
...
         subscriptionForWorkspace(renderLightWorkspaceType({ workspace })),
-        (async () => {
-          return (
-            await FeatureFlag.findAll({
-              where: {
-                workspaceId: workspace?.id,
-              },
-            })
-          ).map((flag) => flag.name);
-        })(),
```
Comment (Fraggle) [inline front/lib/auth.ts:252]: we were quite creative to fetch the feature flag :)

version 1: the anonymous async function call

File: front/lib/auth.ts [hunk 2]
```diff
@@ -299,11 +277,6 @@ export class Authenticator {
           workspace: renderLightWorkspaceType({ workspace }),
         }),
         subscriptionForWorkspace(renderLightWorkspaceType({ workspace })),
-        FeatureFlag.findAll({
-          where: {
-            workspaceId: workspace.id,
-          },
-        }).then((flags) => flags.map((flag) => flag.name)),
```
Comment (Fraggle) [inline front/lib/auth.ts:306]: version 2: the then pattern

File: front/lib/auth.ts [hunk 3]
```diff
@@ -425,32 +390,21 @@ export class Authenticator {
...
-        // Key related attributes.
-        GroupResource.listWorkspaceGroupsFromKey(key),
-        requestedGroupIds
-          ? GroupResource.listGroupsWithSystemKey(key, requestedGroupIds)
-          : [],
-        getSubscriptionForWorkspace(keyWorkspace),
-        getFeatureFlags(keyWorkspace),
-        // Workspace related attributes.
-        getSubscriptionForWorkspace(workspace),
-        getFeatureFlags(workspace),
```
Comment (Fraggle) [inline front/lib/auth.ts:452]: version 3 : the dedicated function

**Review by spolu — APPROVED** [review]
Looks great! Let's see if it saves us some cost 👍👍

File: front/pages/api/w/[wId]/feature-flags.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    WithAPIErrorResponse<GetWorkspaceFeatureFlagsResponseType>
+  >,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.getNonNullableWorkspace();
+
+  switch (req.method) {
+    case "GET":
+      const feature_flags = await getFeatureFlags(owner.id);
+      return res.status(200).json({ feature_flags });
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/feature-flags.ts:28]: `nit` on the private API we use camelCase.

File: front/components/poke/features/table.tsx [hunk 5]
```diff
@@ -29,6 +30,7 @@ export function FeatureFlagsDataTable({
   whitelistableFeatures,
 }: FeatureFlagsDataTableProps) {
   const router = useRouter();
+  const { featureFlags } = useFeatureFlags({ workspaceId: owner.sId });
```
Comment (flvndvd) [inline front/components/poke/features/table.tsx:33]: We try to have poke uses its own endpoints.

**Comment by flvndvd** [comment]
🔥 🔥

File: front/pages/api/w/[wId]/feature-flags.ts (hunk 4 above)
Comment (spolu) [reply front/pages/api/w/[wId]/feature-flags.ts:28]: Yes you're right let's edit this 👍

**Pull request merged**